data-prepper-script/src/main/java/org/opensearch/dataprepper/script/ScriptEngine.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.script;
+
+import com.amazon.dataprepper.model.event.Event;
+
+/**
+ * @since 1.3
+ * ScriptEngine interface to abstract the parse and evaluate implementations.
+ */
+public interface ScriptEngine {
```

Comment:
This seems to be quite a bit of scope creep. And it adds security implications which we may not want to add.

The current goal is to support expression evaluation. Maybe there would eventually be overlap with scripting, but that overlap need not exist.

I'd like to use a name that is specifically tied to expressions and expression evaluation.